torchgen/dest/lazy_ir.py

```diff
@@ -142,29 +157,53 @@ def node_base_ctor_call(self, schema: LazyIrSchema) -> str:
         base_ctor_value_args = ", ".join(base_ctor_value_args_list)
 
         scalar_args = schema.filtered_args(values=False, scalars=True)
-        scalar_hashes = ", ".join([f"{a.name}" for a in scalar_args])
 
-        return f"""{self.node_base}(torch::lazy::OpKind({aten_symbol(schema)}),
-              {{{base_ctor_value_args}}}, std::move(shapes),
+        # Shape constuction.
+        # Conditionally build shape depending on whether it is a native function
+        # and whether the shape cache should be used
```

Comment:
what is the reason to avoid using the shape cache for certain ops?

if it is to align with what is currently happening on master, then I would think we would want to take advantage of doing the codegen and use the shape cache globally.

Reply:
Its to make the hash function unique. I was seeing that caching the shapes for some of the non-native IR nodes (and thus not using them to calculate the hash) was causing incorrect cache hits and thus causing errors. It was causing a whole bunch of tests to fail.

Fortunately, the shapes that aren't being cached (marked in `ts_native_functions.yaml`) have trivial shape functions and thus don't really need or benefit from shape cachine

Reply:
Do you have examples for those failures? I want to understand why they are failing, and the least we can add them as tests.

Reply:
I'm not entirely sure why they specifically are failing. I think likely to do with the fact the shapes are inherent the node's uniqueness. Without which they get "mistaken" for other similar nodes due to false shape cache hits.

For now, I think just keeping the behaviour the same as it was before should be sufficient as changing those does already lead to test failures. Thoughts?